### DISCUSSION TITLE
With multiple windows, which window gets to open a file?

### DISCUSSION THREAD

**Discussion by gnuarm** [sumatrapdfreader/sumatrapdf#2314] [Q&A]
I open multiple windows so I can group documents for various topics making access easier.  However, it limits my choices for how to open documents in all the windows but one.  The one window (I suppose the original window) will open the document if I double click in Windows explorer or any other means such as from a browser.  The other windows will only open a document if I use the open feature within the window.  

I would have expected the most recently used window or the window in display (not minimized) to open the document.  Is there any way for me to double click a document in Windows Explorer or drag it to a Sumatra window to open it in THAT window?

**Comment by kjk** [comment]
Not currently but it makes sense. I've opened https://github.com/sumatrapdfreader/sumatrapdf/issues/2315 to track this.

**Comment by kjk** [comment]
I fixed it so in about an hour the fix should be available in [latest pre-release versoin](https://www.sumatrapdfreader.org/prerelease) 14262 or later

**Reply by gnuarm** [reply]
I'm not clear what the fix involves.  I downloaded ver 14267 and it seems to work approximately the same.  Seems like both the old version 3.3.3 and this new one always open a file in the last Sumatra window opened when dragged from a Windows Explorer window to any Sumatra window.  

I was hoping I could control the destination window when multiple windows are open.  

Also, some previous version I've been using would restore multiple windows when started.  This prerelease version and 3.3.3 only open one window from the previous session and none of the others.  If closed through the app, the window opened when started is the first window opened from the previous session.  If the application is crashed, the one window opened is the first window opened with the new window menu item.

**Comment by kjk** [comment]
You said:
> I would have expected the most recently used window or the window in display (not minimized) to open the document

That is the new behavior: when you double-click or run cmd-line `SumatraPDF.exe <file>`, the file is opened in the last visible window. 

If you drag&drop, it opens in the window on which you dropped.

 > I was hoping I could control the destination window when multiple windows are open.

Control how?

As to your second question: please only ask one question per thread to help people in the future find the information.

To answer it: with default settings, if you quit with File/Exit (Ctrl-Q) (as opposed to, e.g. with Alt-F4), all windows will be restored.

**Reply by gnuarm** [reply]
This is a bit hard to test, but I must have had something goofed up yesterday.  Today the file always opens in the most recent window accessed.  Even if I drag and drop to another window, it opens in the window that was most recently in focus.

**Reply by kjk** [reply]
I fixed the drag&drop issue in https://github.com/sumatrapdfreader/sumatrapdf/issues/2323

**Comment by aleyougu** [comment]
thxs, I was going to ask this question and and find the answer here. :-)
